### PR TITLE
V1.3 add editleave command

### PR DESCRIPTION
**Pull request by inezkok**
Added relevant classes (EditLeaveCommand, EditLeaveCommandParser)
Not done with testcases (still working on EditLeaveCommandTest, done with EditLeaveCommandParserTest)

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditLeaveCommand.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditLeaveCommand)) {
+            return false;
+        }
+
+        EditLeaveCommand otherEditLeaveCommand = (EditLeaveCommand) other;
+        return targetId.equals(otherEditLeaveCommand.targetId);
```
Comment (LordSaumya) [inline src/main/java/seedu/address/logic/commands/EditLeaveCommand.java:128]: Shouldn't the dates for the two editLeaveCommands also be compared to check if they are duplicates?

File: src/test/java/seedu/address/logic/parser/EditLeaveCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    public void parse_invalidValue_failure() {
+        // invalid employee id
+        String userInput = INVALID_ID_DESC + " " + PREFIX_OLD + "2023-10-10 " + PREFIX_NEW + "2023-10-11";
+        assertParseFailure(parser, userInput, MESSAGE_INVALID_FORMAT);
+
+        // invalid old date
+        userInput = " " + PREFIX_ID + VALID_ID_BOB + " " + PREFIX_OLD + " " + PREFIX_NEW + "2023-10-11";
+        assertParseFailure(parser, userInput, Leave.MESSAGE_CONSTRAINTS);
+
+        // invalid end date
```
Comment (LordSaumya) [inline src/test/java/seedu/address/logic/parser/EditLeaveCommandParserTest.java:52]: Did you mean "invalid new date"?

File: src/test/java/seedu/address/logic/parser/EditLeaveCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Test
+    public void parse_allFieldsSpecified_success() {
+        String userInput = " " + PREFIX_ID + VALID_ID_BOB + " " + PREFIX_OLD
+                + VALID_LEAVELIST_BOB.get(0).toString() + " " + PREFIX_NEW + "2023-11-11";
+        EditLeaveCommand expectedCommand = new EditLeaveCommand(new Id(VALID_ID_BOB), LocalDate.now(),
+                LocalDate.of(2023, 11, 11));
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+}
```
Comment (LordSaumya) [inline src/test/java/seedu/address/logic/parser/EditLeaveCommandParserTest.java:65]: Nice job on the testing, Maybe also check if new date is the same as the old date?

**Review by miljyy — APPROVED** [review]
Well done implementing the command and tests!

**Pull request merged**